probs/probs.go

```diff
@@ -142,3 +144,34 @@ func ContentLengthRequired() *ProblemDetails {
 		HTTPStatus: http.StatusLengthRequired,
 	}
 }
+
+// RollbackError is a combination of a database error and the error, if any,
+// encountered while trying to rollback the transaction.
+type RollbackError struct {
+	Err         error
+	RollbackErr error
+}
+
+// Error implements the error interface
+func (re *RollbackError) Error() string {
+	if re.RollbackErr == nil {
+		return re.Err.Error()
+	}
+	return fmt.Sprintf("%s\nWhile rolling back the database: %s", re.Err, re.RollbackErr)
+}
+
+// WithRollbackError rolls back the provided transaction (if err is non-nil)
+// and wraps the error, if any, of the rollback into a RollbackError.
+//
+// If err is nil, the transaction will NOT be rolled back.
```

Comment:
With the changes I describe above (only calling Rollback if err is non-nil), we can remove the semantics of not rolling back when err is nil. I think that makes the functionality a bit clearer.